migrations/sql/V2019.03.12_1523__InitialSchema.sql

```diff
@@ -0,0 +1,46 @@
+CREATE TABLE "orderlyweb_user" (
+"username"  TEXT ,
+"email" TEXT ,
+PRIMARY KEY ("username")
+);
+
+CREATE TABLE "orderlyweb_user_group" (
+"id"  TEXT ,
+PRIMARY KEY ("id")
+);
+
+CREATE TABLE "orderlyweb_user_group_user" (
+"user"  TEXT NOT NULL,
+"user_group" TEXT NOT NULL,
+FOREIGN KEY ("user") REFERENCES "orderlyweb_user" ("username") ,
+FOREIGN KEY ("user_group") REFERENCES "orderlyweb_user_group" ("id")
+);
+
+CREATE TABLE "orderlyweb_permission" (
+"id"  TEXT ,
+PRIMARY KEY ("id")
+);
+
+CREATE TABLE "orderlyweb_user_group_permission" (
```

Comment:
The permissions table are using the pattern we talked about, with a 'base' table holding the ids and basic info for all perms, with further tables to hold any specialisations (for report and version perms)

Reply:
Yeah, I didn't think enough about global permissions here. I think I just had it in mind that if a row in the user_group_permission table didn't join against any of scoped tables (report or version) then it must be global, but that's really too error prone. I'd prefer to add the user_group_global_permission table with FK into the ID of the base table than to have the added scope column, as it would be consistent with how the other scopes work, it just wouldn't have any additional information to keep in any other columns. 

The argument for having the 'base' table is that the shared columns only need to defined in one place, which feels cleaner (e.g. if the permission id column were to change from text to integer we'd only have to update a foreign key in one table rather than three). And yes, it means you can re-use the rows in the base column with different scopes. I think I still prefer this approach, if we added user_group_global_permission.

Reply:
Ok, lets go with this way :+1: